backend/filen/filen.go

```diff
@@ -0,0 +1,1084 @@
+		err = listHelper.Add(&Directory{
+			fs:        f,
+			directory: dir,
+			path:      path,
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, file := range files {
+		parentPath, err := getPathForUUID(file.GetParent(), uuidPathMap, uuidDirMap)
+		if err != nil {
+			return err
+		}
+		err = listHelper.Add(&File{
+			fs:   f,
+			file: file,
+			path: pathModule.Join(parentPath, file.GetName()),
+		})
+		if err != nil {
+			return err
+		}
+	}
+	return listHelper.Flush()
+}
+
+func buildUUIDDirMaps(rootPath string, rootDir types.DirectoryInterface, dirs []*types.Directory) (map[string]types.DirectoryInterface, map[string]string) {
+	uuidPathMap := make(map[string]string, len(dirs)+1)
+	uuidPathMap[rootDir.GetUUID()] = rootPath
+
+	uuidDirMap := make(map[string]types.DirectoryInterface, len(dirs)+1)
+	uuidDirMap[rootDir.GetUUID()] = rootDir
+	for _, dir := range dirs {
+		uuidDirMap[dir.GetUUID()] = dir
+	}
+	return uuidDirMap, uuidPathMap
+}
+
+func getPathForUUID(uuid string, uuidPathMap map[string]string, uuidDirMap map[string]types.DirectoryInterface) (string, error) {
+	if path, ok := uuidPathMap[uuid]; ok {
+		return path, nil
+	}
+	dir, ok := uuidDirMap[uuid]
+	if !ok {
+		return "", fs.ErrorDirNotFound
+	}
+	parentPath, err := getPathForUUID(dir.GetParent(), uuidPathMap, uuidDirMap)
+	if err != nil {
+		return "", err
+	}
+	path := pathModule.Join(parentPath, dir.GetName())
+	uuidPathMap[uuid] = path
+	return path, nil
+}
+
+// About gets quota information from the Fs
+func (f *Fs) About(ctx context.Context) (*fs.Usage, error) {
+	userInfo, err := f.filen.GetUserInfo(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	total := int64(userInfo.MaxStorage)
+	used := int64(userInfo.UsedStorage)
+	free := total - used
+	return &fs.Usage{
+		Total:   &total,
+		Used:    &used,
+		Trashed: nil,
+		Other:   nil,
+		Free:    &free,
+		Objects: nil,
+	}, nil
+}
+
+// CleanUp the trash in the Fs
+func (f *Fs) CleanUp(ctx context.Context) error {
+	// not sure if this is implemented correctly, since this trashes ALL trash
+	// not just the trash in the currently mounted fs
+	// not currently wiping file versions because that feels dangerous
+	// especially since versioning can be toggled on/off
+	return f.filen.EmptyTrash(ctx)
+}
+
+// helpers
+
+// resolvePath returns the absolute path specified by the input path, which is seen relative to the remote's root.
+func (f *Fs) resolvePath(path string) string {
+	return pathModule.Join(f.root.path, path)
+}
+
+// Check the interfaces are satisfied
+var (
+	_ fs.Fs          = &Fs{}
+	_ fs.Mover       = &Fs{}
+	_ fs.DirMover    = &Fs{}
+	_ fs.Purger      = &Fs{}
+	_ fs.PutStreamer = &Fs{}
+	_ fs.CleanUpper  = &Fs{}
+	_ fs.ListRer     = &Fs{}
+	_ fs.Abouter     = &Fs{}
+	_ fs.Directory   = &Directory{}
+	_ fs.Object      = &File{}
+	_ fs.MimeTyper   = &File{}
+	_ fs.IDer        = &File{}
+	_ fs.ParentIDer  = &File{}
+)
+
+// todo PublicLinker,
+// we could technically implement ChangeNotifier, but
```

Comment:
ChangeNotify is great for long running `rclone mount`. Yes rclone's implementation is very lowest common denominator - it is just use to flush the directory cache when we know a directory has changed.